tests/schemabuilder/builder_test.py

```diff
@@ -16,227 +18,219 @@
-        assert self.schema.openapi
-        assert self.schema.info
-        assert self.schema.paths
-
-    def test_paths(self):
-        paths = self.schema.paths
-        assert '/user/repos' in paths
-
-    def test_get_operation(self):
-        op = self.schema.paths['/user/repos'].get
-        assert op.responses
-
-    def test_get_operation_responses(self):
-        responses = self.schema.paths['/user/repos'].get.responses
-        assert '200' in responses
-        assert '403' in responses
-
-    def test_200_response(self):
-        response = self.schema.paths['/user/repos'].get.responses['200']
-        assert 'x-dns-prefetch-control' in response.headers
-        assert response.links == {}
-        assert 'application/json' in response.content
-        media_type = response.content['application/json']
-        assert media_type.schema
-
-    def test_repos_content(self):
-        content = self.schema.paths['/user/repos'].get.responses['200'].content
-        assert_that(content, has_key("application/json"))
-        json_content = content['application/json']
-        assert json_content.schema
-        schema = json_content.schema
-        assert schema._type == "array"
-        # TODO Typeguard for Reference
-        assert isinstance(schema.items, Schema)
-
-    def test_items(self):
-        schema = self.response_schema()
-        items = schema.items
-        assert isinstance(items, Schema)  # typeguard
-        assert isinstance(items.properties, dict)
-        properties = items.properties
-        assert_that(properties, has_entry('clone_url',
-                                          equal_to(Schema(**{**_SCHEMA_DEFAULT, '_type': "string"}))))
+#### petstore
 
-    def test_servers(self):
-        servers = self.schema.servers
-        assert 1 == len(servers)
-        assert 'http://api.github.com' == servers[0].url
+get_pets_req = RequestBuilder.from_url(
+    "http://petstore.swagger.io/v1/pets")
+get_one_pet_req = RequestBuilder.from_url(
+    "http://petstore.swagger.io/v1/pets/32")
 
-    def test_pokeapi_schema_valid(self):
-        # this should conflate to
-        # /pokemon
-        # /abilities/*
-        # /types/*
-        # /pokemon/*
-        # meaning that it should recognize wildcards
-        # from all these paths
-        paths = self.pokeapi_schema.paths.keys()
-        assert 4 == len(paths)
-        assert_that(paths, has_item("/v2/pokemon/"))
-        assert_that(paths, has_item(
-            matches_regexp(r'\/v2\/pokemon\/\{[\w]+\}')))
-        assert_that(paths, has_item(
-            matches_regexp(r'\/v2\/type\/\{[\w]+\}')))
-        assert_that(paths, has_item(
-            matches_regexp(r'\/v2\/ability\/\{[\w]+\}')))
+pet_res = Response(bodyAsJson=None, timestamp=None, body="", statusCode=200, headers={})
 
+@pytest.fixture
+def get_one_pet_exchange():
+    return HttpExchange(request=get_one_pet_req, response=pet_res)
 
-class TestPetstoreSchemaUpdate:
+@pytest.fixture
```

Comment:
You didn't like grouping tests for some reason? That's fine, just wondering why.

Reply:
We could keep the grouping if you'd like, but what I didn't like was the building of the resource outside of the test. The problem is that if you're trying to do debugging (ie spewing stuff to the console) it does not colocate the debug messages with the test, which makes debugging challenging.